src/big/big_float.cr

```diff
@@ -18,7 +18,39 @@ struct BigFloat < Float
   end
 
   def initialize(num : Number)
-    LibGMP.mpf_init_set_d(out @mpf, num.to_f64)
+    # XXX: this case is workaround of Crystal's unrealiable method overloading
+    # remove it when separate BigFloat.new(BigInt) will pass the spec
+    case num
+    when BigInt
+      LibGMP.mpf_init(out @mpf)
```

Comment:
I forgot this PR depended on https://github.com/crystal-lang/crystal/pull/4653. We should really get that PR merged so that this PR can be rebased on top of it.